src/diart/models.py

```diff
@@ -1,29 +1,45 @@
 from abc import ABC, abstractmethod
-from typing import Optional, Text, Union, Callable
+from typing import Optional, Text, Union, Callable, Mapping, TYPE_CHECKING
```

Comment:
`Mapping` and `TYPE_CHECKING` should be removed

Reply:
I removed Mapping, I'm not 100% sure about the TYPE_CHECKING though. Currently if no pyannote is available we're still using `Union[Model, PretrainedSpeakerEmbedding]` as a type. As such, if we're type checking and pyannote is unavailable we are failing because `Model` is not known. If we keep TYPE_CHECKING, we would fail with ImportError. Nonetheless I can remove the TYPE_CHECKING block. Please just verify that you still want me to.

Reply:
well, if pyannote is installed, we need them for `Model.from_pretrained()` and `PretrainedSpeakerEmbedding(model_info)`, so not a type checking usage.
If pyannote isn't installed, then the code shouldn't fail. I think in this case we need to change `Union[Model, PretrainedSpeakerEmbedding]` to `Callable` for that to work.

In both cases, we can remove the `TYPE_CHECKING` import